plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidRemoteIdentifierSupplier.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.druid;
+
+import io.trino.plugin.jdbc.mapping.RemoteIdentifierSupplier;
+import io.trino.spi.TrinoException;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.SQLException;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+
+public class DruidRemoteIdentifierSupplier
+        implements RemoteIdentifierSupplier
+{
+    @Override
+    public String toRemoteIdentifier(Connection connection, String identifier)
+    {
+        try {
+            DatabaseMetaData metadata = connection.getMetaData();
```

Comment:
Let's not call this method if not needed. It may populate a real connection in `LazyConnection`